sagan-common/src/main/java/sagan/support/cache/CachedRestClient.java

```diff
@@ -17,6 +18,11 @@
         return operations.getForObject(url, clazz);
     }
 
+    @Cacheable(value = CACHE_NAME, key = "#url")
```

Comment:
This won't work since we're already using that key to cache objects of types `T`. So fetching from the cache using an URL, you can get the wrong thing depending on the method.

Also, `ResponseEntity` is not serializable and that won't work with the default JDK serializer used by the redis cache manager. See `CacheConfig` for the configuration used for other types.